tests/app/main/test_login.py

```diff
@@ -305,14 +409,724 @@ def test_reset_password_form_and_inputs_not_autofillable(
+            assert_equal(
+                "Check you&#146;ve entered the correct link or ask the person who invited you to send a new invitation." in res.get_data(as_text=True),  # noqa
+                True
+            )
+            assert_equal(
+                '<button class="button-save">Create contributor account</button>' in res.get_data(as_text=True),
+                False
+            )
+
+    def test_should_be_an_error_if_invalid_token_on_update(self):
+        with self.app.app_context():
+            res = self.client.post(
+                '/suppliers/update-user/invalidtoken'
+            )
+
+            assert_equal(res.status_code, 400)
+            assert_equal(
+                "Check you&#146;ve entered the correct link or ask the person who invited you to send a new invitation." in res.get_data(as_text=True),  # noqa
+                True
+            )
+            assert_equal(
+                '<button class="button-save">Update user</button>' in res.get_data(as_text=True),  # noqa
+                False
+            )
+
+    def test_should_be_an_error_if_missing_name_and_password(self):
+        with self.app.app_context():
+
+            token = generate_token(
+                {
+                    'supplier_id': 1234,
+                    'supplier_name': 'Supplier Name',
+                    'email_address': 'testme@email.com'
+                },
+                self.app.config['SHARED_EMAIL_KEY'],
+                self.app.config['INVITE_EMAIL_SALT']
+            )
+
+            res = self.client.post(
+                '/suppliers/create-user/{}'.format(token),
+                data={
+                    'password': None,
+                    'name': None
+                }
+            )
+
+            assert_equal(res.status_code, 400)
+            assert_equal(
+                "Please enter a password" in res.get_data(as_text=True),
+                True
+            )
+            assert_equal(
+                "Please enter a name" in res.get_data(as_text=True),
+                True
+            )
+
+    def test_should_be_an_error_if_too_short_name_and_password(self):
+        with self.app.app_context():
+
+            token = generate_token(
+                {
+                    'supplier_id': 1234,
+                    'supplier_name': 'Supplier Name',
+                    'email_address': 'testme@email.com'
+                },
+                self.app.config['SHARED_EMAIL_KEY'],
+                self.app.config['INVITE_EMAIL_SALT']
+            )
+
+            res = self.client.post(
+                '/suppliers/create-user/{}'.format(token),
+                data={
+                    'password': "123456789",
+                    'name': ""
+                }
+            )
+
+            assert_equal(res.status_code, 400)
+            assert_equal(
+                "Please enter a name" in res.get_data(as_text=True),
+                True
+            )
```

Comment:
fixed. *guilty face* been doing that a bit, will remember and fix 'em when I see 'em